src/storage/syncer/stateMutators.ts

```diff
@@ -1497,6 +1509,99 @@ export function mergeAppInstallation(
     return currAppState;
 }
 
+export function setTask(currAppState: AppState, task: Task, taskLinks?: TaskLink[], attachmentList?: AttachmentList, attachments?: Attachment[]): AppState {
```

Comment:
```suggestion
export function setTask(currAppState: AppState, task: Task): AppState {
```
Can we decouple 
```
taskLinks?: TaskLink[], attachmentList?: AttachmentList, attachments?: Attachment[]
```
out of this API?